go/grpcweb/websocket_wrapper.go

```diff
@@ -10,24 +11,67 @@ import (
 	"net/http"
 	"net/textproto"
 	"strings"
+	"time"
 
 	"github.com/gorilla/websocket"
 	"golang.org/x/net/http2"
 )
 
 type webSocketResponseWriter struct {
-	writtenHeaders bool
-	wsConn         *websocket.Conn
-	headers        http.Header
-	flushedHeaders http.Header
+	writtenHeaders  bool
+	wsConn          *websocket.Conn
+	headers         http.Header
+	flushedHeaders  http.Header
+	timeOutInterval uint
+	tickerCount     uint
```

Comment:
您好，我尝试使用lastMessageTime代替tickerCount来处理校验是否超时发送Ping。但是使用ticker的思路导致会频繁调用ticker的超时函数进行额外的计算，而且精度会受ticker间隔大小的影响，所以我尝试了使用timer.Reset()方法来重置定时器。目前看起来工作是正常的，但是不知道这样实现是否有未考虑到的隐患。
Hello, I am trying to use lastMessageTime instead of tickerCount to handle whether the check is timed to send a ping. But the idea of using ticker leads to frequent calls to ticker's timeout function for additional calculations, and the precision is affected by the size of the ticker interval, so I tried to reset the timer using the timer.Reset() method. At the moment it seems that the work is normal, but I don't know if there are hidden dangers in this implementation.
（Translations courtesy of google translate）